### PR TITLE
fix(fmt, lint): Make sure that target paths are not directory

### LINKED ISSUE #8374: regression: deno fmt directory matching fails
Running deno fmt in this repo: https://github.com/lucacasonato/dext.ts, causes an error since some time after 1.5.2. Possibly related to #8040. I have not been able to narrow it down yet.

```
> deno fmt
error: Is a directory (os error 21)
```

### PR DESCRIPTION
**Pull request by magurotuna**
Fixes #8374 

~~I'll fix more and add tests, so WIP for now.~~ Done

### CODE REVIEW

**Comment by bartlomieju** [comment]
@magurotuna huge thanks for looking into this! I noticed that in `fs.rs` there is `files_in_subtree` method used in a few places. Do you think we could combine the two implementations into a single one?

**Comment by magurotuna** [comment]
@bartlomieju Oh, I missed `files_in_subtree`, thanks for pointing out. It seems usable in this fix. I'll try it tomorrow :)

**Comment by magurotuna** [comment]
@bartlomieju PTAL😉

File: cli/fs.rs [hunk 1]
```diff
@@ -118,18 +176,83 @@ mod tests {
...
+    // ├── child
+    // │   ├── e.mjs
+    // │   ├── f.mjsx
+    // │   ├── .foo.TS
+    // │   └── README.md
+    // ├── c.tsx
+    // ├── d.jsx
+    // └── ignore
+    //     ├── g.d.ts
+    //     └── .gitignore
```
Comment (bartlomieju) [inline cli/fs.rs:219]: 👍

**Review by bartlomieju — APPROVED** [review]
I verified that problem described in https://github.com/denoland/deno/issues/8374 is no longer present. Great fix and cleanup @magurotuna, thanks!

**Pull request merged**